nixos/modules/services/misc/gitea.nix

```
@@ -10,25 +10,32 @@ let
   useMysql = cfg.database.type == "mysql";
   usePostgresql = cfg.database.type == "postgres";
   useSqlite = cfg.database.type == "sqlite3";
-  format = pkgs.formats.ini { };
-  configFile = pkgs.writeText "app.ini" ''
-    APP_NAME = ${cfg.appName}
-    RUN_USER = ${cfg.user}
-    RUN_MODE = prod
 
-    ${generators.toINI {} cfg.settings}
+  format = pkgs.formats.ini {
+    globalSection = true;
+  };
 
-    ${optionalString (cfg.extraConfig != null) cfg.extraConfig}
-  '';
+  configFile = pkgs.writeText "app.ini" (
+    format.generateText cfg.settings
+    + lib.optionalString (cfg.extraConfig != null && extraConfig != "") "\n${extraConfig}"
+  );
```

I don't think `generateText` is a good idea for the formats, but it's also not needed, because you can just do
```suggestion
  configFile = pkgs.runCommand "app-extra.ini" {
    generated = format.generate "app.ini" cfg.settings;
    inherit extraConfig;
    passAsFile = [ "extraConfig" ];
  } ''
    cat "$generated" "$extraConfigPath"
  ''
```

(might need to add an extra newline in there)

---

`generateText` _is_ relevant for the trusted evaluator kind of secrets deployment that nixops supports. That's not how it's used here though.

---

IMO using `pkgs.runCommand` for this is a bit weird but it's not a strong opinion, so I'll update the MR.

---

We could also implement a `pkgs.concatFiles [ a b ]` which would just do `cat ${a} ${b}` and put the result in a new derivation. The fundamental idea of only reading the file at build time is key, since we don't actually need the string at eval-time.

---

@roberth I think the secret content should be concatenated using `cat` at runtime then, still not during evaluation